skll/data/readers.py

```diff
@@ -454,6 +472,9 @@ def read(self):
             print(self._progress_msg, end="\r", file=sys.stderr)
             sys.stderr.flush()
 
+        # if we are in this method, self.path_or_file must be a path
+        assert isinstance(self.path_or_list, (str, Path)), "file path or path object required"
+
```

Comment:
why can't we convert it to a Path?

Reply:
Oh I see it's a dict.

Reply:
usually we prefer raise error over assert, right?

Reply:
Yes, we do. Let me try and refactor this a bit.

Reply:
Actually, this assert statement will raise an `AssertionError` if it fails so it's the same thing.